### PR TITLE
Handle docker restart

### PR DESCRIPTION
**Pull request by amitlimaye**
#### Description
*Changes proposed in this pull request.*

#### Test plan
*Outline the test plan used to test this change before merging it.*

> Fixes #.

### CODE REVIEW

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "ab297930df33779893b6c060a256eba5a8d6fcbf"
  }
]
```

File: monitor/internal/docker/monitor.go [hunk 1]
```diff
@@ -221,17 +221,34 @@ func (d *DockerMonitor) eventProcessors(ctx context.Context) {
...
+			// We are here means the docker daemon restarted. we need to resync
+			if err := d.Resync(ctx); err != nil {
+				zap.L().Error("Unable to resync containers after reconnecting to docker daemon", zap.Error(err))
+			}
+		}
+		d.listener(ctx)
+
+	}
+}
+
```
Comment (dstiliadis) [inline monitor/internal/docker/monitor.go:267]: I think it needs a lock. There are two go-routines fighting for the d.dockerClient

**Review by dstiliadis — CHANGES_REQUESTED** [review]
see inline

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "70225a007afaf6b754ad4b55400181fd0e0ef272"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "d63b210c3414f55b3ba4ea9e872ca7cd7ab9eed5"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "0022404bd073d2e62b7f4a7495c54d882962acb6"
  }
]
```

File: monitor/internal/docker/monitor.go (hunk 1 above)
Comment (amitlimaye) [reply monitor/internal/docker/monitor.go:267]: Added a lock around accessing dockerClient

**Review by dstiliadis — DISMISSED** [review]

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "a06a3bbf10ff8b89b8fdfd98bf88829a8e6ea4e2"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "a06a3bbf10ff8b89b8fdfd98bf88829a8e6ea4e2"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "7519c68e55d27763c14a937ce9a7a474cdf4738f"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "0022404bd073d2e62b7f4a7495c54d882962acb6"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "592823304fc2e6082199db5162f2237a3442038f"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "f4e3e03b361f0615010783201f7e07c15efb5589"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "aec418776853c0dd04def2a2d434409d3d11bd9f"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "c6768d358c6b9ee111d065bc4c28c7c7e1a79f4a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "5ca8665fd61d7cb49ce92a7fca78b513602529e6"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "6e64f48d70c12a34ad6e29bdb00a8587a0afcbee"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "5ca8665fd61d7cb49ce92a7fca78b513602529e6"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "6e64f48d70c12a34ad6e29bdb00a8587a0afcbee"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "c388cb8b239bedc6dcd9d005b32d317ba0bbbbb8"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "6e64f48d70c12a34ad6e29bdb00a8587a0afcbee"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "2b725a774dcff67785136d30100de2950dc3ea9c"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "3630c77596e2f04abad3d2ee566cbcde3b01932c"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "4c526fd74f203f518b7b0642d56eea234fee996b"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "3630c77596e2f04abad3d2ee566cbcde3b01932c"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "4c526fd74f203f518b7b0642d56eea234fee996b"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1241",
    "commit-sha": "e18f7d774cc9d731d80ec20fabacabdaf9344607"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "833",
    "commit-sha": "5fba15c7ba2521ee5e379fabb23e4e83922da0f3"
  }
]
```

**Pull request merged**